### PR TITLE
add graal vm native image support for jhipster lite

### LINKED ISSUE #2819: Add GraalVM support for JHipster Lite

### PR DESCRIPTION
**Pull request by atomfrede**
Running following works fine now:

```
$ ./mvnw native:compile -Pnative -DskipTests
$ ./target/jhlite
```

I needed to add native hints for some of our rest classes to be correctly serialized via jackson in the native image.
The most important parts are in the common module which configures jgit and dom4j to work within the native image.

closes #2819

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/common/infrastructure/secondary/NativeHints.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      .registerType(CoreConfig.EOL.class, MemberCategory.values())
+      .registerType(CoreConfig.EolStreamType.class, MemberCategory.values())
+      .registerType(CoreConfig.HideDotFiles.class, MemberCategory.values())
+      .registerType(CoreConfig.SymLinks.class, MemberCategory.values())
+      .registerType(CoreConfig.LogRefUpdates.class, MemberCategory.values());
+
+    // Needed for xml reading and writing via dom4j
+    hints
+      .reflection()
+      .registerType(TypeReference.of("org.dom4j.util.ProxyDocumentFactory"), MemberCategory.values())
```
Comment (atomfrede) [inline src/main/java/tech/jhipster/lite/common/infrastructure/secondary/NativeHints.java:37]: We could use the classes directly here in fact.

**Comment by atomfrede** [comment]
Will need to run local sonar from my linux machine later today

**Comment by atomfrede** [comment]
@DamnClin Just to get your initial feedback regarding the hexagonal architecture. I put some of the native hint configuration into common (3rd party classes). Other configs which use e.g. some code from our modules into the respective modules. Do you consider this fine or should we structure it differently?

Regarding test coverage, what would you say how to test these native hint configurations? I think doing a unit test mocking some stuff seems quite useless (except fulfilling the coverage requirements).

File: pom.xml [hunk 2]
```diff
@@ -6,6 +6,12 @@
 >
   <modelVersion>4.0.0</modelVersion>
 
+  <parent>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-parent</artifactId>
+    <version>3.0.0-RC1</version>
```
Comment (DamnClin) [inline pom.xml:12]: versions should be kept in properties

File: src/main/java/tech/jhipster/lite/common/infrastructure/secondary/NativeHints.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package tech.jhipster.lite.common.infrastructure.secondary;
+
+import org.dom4j.*;
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/common/infrastructure/secondary/NativeHints.java:3]: Only static imports should be *

File: src/main/java/tech/jhipster/lite/module/infrastructure/primary/OpenApiModulesConfiguration.java [hunk 4]
```diff
@@ -15,8 +15,7 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import org.springdoc.core.customizers.OpenApiCustomizer;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.*;
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/module/infrastructure/primary/OpenApiModulesConfiguration.java:18]: Only static imports should be *

**Comment by DamnClin** [comment]
> Regarding test coverage, what would you say how to test these native hint configurations? I think doing a unit test mocking some stuff seems quite useless (except fulfilling the coverage requirements).

You can annotate them `@Generated`. The tests are run in the native image during build?

**Comment by atomfrede** [comment]
> > Regarding test coverage, what would you say how to test these native hint configurations? I think doing a unit test mocking some stuff seems quite useless (except fulfilling the coverage requirements).
> 
> You can annotate them `@Generated`. The tests are run in the native image during build?

Right now the tests are not run in a native image/native mode.

File: src/main/java/tech/jhipster/lite/common/infrastructure/secondary/NativeHints.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
+package tech.jhipster.lite.common.infrastructure.secondary;
```
[the remark this answers is not in the document]
Comment (atomfrede) [reply src/main/java/tech/jhipster/lite/common/infrastructure/secondary/NativeHints.java:1]: 🤦 I did that for the rest stuff and totally missed the git package. So in common only the dom4j stuff remains (if I can't find a clear place where it belongs to, maybe the maven module or so)

File: src/main/java/tech/jhipster/lite/common/infrastructure/secondary/NativeHints.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+import org.dom4j.*;
+import org.dom4j.util.*;
+import org.eclipse.jgit.api.*;
+import org.eclipse.jgit.internal.*;
+import org.eclipse.jgit.lib.*;
+import org.springframework.aot.hint.*;
+
+public class NativeHints implements RuntimeHintsRegistrar {
```
[the remark this answers is not in the document]
Comment (atomfrede) [reply src/main/java/tech/jhipster/lite/common/infrastructure/secondary/NativeHints.java:10]: We could create a package private configuration and use it there, right now it is used in the application, but package private would be better

**Comment by DamnClin** [comment]
> Right now the tests are not run in a native image/native mode.

I think it's really important to run the tests in the native image :)

**Comment by atomfrede** [comment]
> > Right now the tests are not run in a native image/native mode.
> 
> I think it's really important to run the tests in the native image :)

Yes. I think we can do this in a second step as it will have a significant impact on ci build times for example

File: pom.xml (hunk 2 above)
Comment (atomfrede) [reply pom.xml:12]: Can't use versions in parent definition

**Comment by atomfrede** [comment]
I tried the native test execution. It seems cucumber is not yet compatible to native testing. So we would need to run cucumber from a normal jdk pointing to a running native image or container.

File: pom.xml (hunk 2 above)
Comment (atomfrede) [reply pom.xml:12]: My solution, use the boot version from parent pom instead dedicated spring-boot.version property

**Comment by pascalgrimaud** [comment]
Wow! Nice work here @atomfrede 
Let me think how we can test it without slowing down the full pipeline

File: pom.xml (hunk 2 above)
Comment (pascalgrimaud) [reply pom.xml:12]: For explaination, it's related to this ticket https://github.com/jhipster/jhipster-lite/issues/579

Comment (atomfrede) [reply pom.xml:12]: Didn't know that issue. Changed it to use the parent to have the native build configurations "out of box"

Comment (pascalgrimaud) [reply pom.xml:12]: Yes, let's use parent like spring initializr

Comment (atomfrede) [inline pom.xml:12]:
```suggestion
    <version>3.0.0-RC2</version>
```

**Comment by pascalgrimaud** [comment]
Sorry, I upgraded the main branch, so you have some conflict here :p

**Comment by atomfrede** [comment]
No problem.

**Comment by pascalgrimaud** [comment]
I tried to follow the README to build a Docker image and it doesn't work:

```
./mvnw spring-boot:build-image -Pnative -DskipTests
```

```
[INFO] --- spring-boot-maven-plugin:3.0.0-RC2:build-image (default-cli) @ jhlite ---
[INFO] Building image 'docker.io/library/jhlite:0.22.1-SNAPSHOT'
[INFO] 
[INFO]  > Pulling builder image 'docker.io/paketobuildpacks/builder:tiny' 100%
[INFO]  > Pulled builder image 'paketobuildpacks/builder@sha256:84a20510c7c991e168019cf2d2e47fda07e44de40084fc3bd7edb740aeb35e31'
[INFO]  > Pulling run image 'docker.io/paketobuildpacks/run:tiny-cnb' 100%
[INFO]  > Pulled run image 'paketobuildpacks/run@sha256:cedde05847df71cf85b035edaece87436f73c72d934914cccaffc469d57d544b'
[INFO]  > Pulling buildpack image 'gcr.io/paketo-buildpacks/bellsoft-liberica:9.9.0-ea' 100%
[INFO]  > Pulled buildpack image 'gcr.io/paketo-buildpacks/bellsoft-liberica@sha256:32e69e3fcc17725ed8a7bddd9d2881a0f1019354a0e653521dac55898c0e8c36'
[INFO]  > Pulling buildpack image 'gcr.io/paketo-buildpacks/java-native-image:latest' 0%
[INFO]  > Pulling buildpack image 'gcr.io/paketo-buildpacks/java-native-image:latest' 23%
[INFO]  > Pulling buildpack image 'gcr.io/paketo-buildpacks/java-native-image:latest' 32%
[INFO]  > Pulling buildpack image 'gcr.io/paketo-buildpacks/java-native-image:latest' 39%
[INFO]  > Pulling buildpack image 'gcr.io/paketo-buildpacks/java-native-image:latest' 40%
[INFO]  > Pulling buildpack image 'gcr.io/paketo-buildpacks/java-native-image:latest' 100%
[INFO]  > Pulled buildpack image 'gcr.io/paketo-buildpacks/java-native-image@sha256:6347a2a1fb05c4fa0767e702246e0d04ff5cc8fd0d80fb445ebd6f7cbe395537'
[INFO]  > Executing lifecycle version v0.15.1
[INFO]  > Using build cache volume 'pack-cache-cbca7a44d7ec.build'
[INFO] 
[INFO]  > Running creator
[INFO]     [creator]     ===> ANALYZING
[INFO]     [creator]     Previous image with name "docker.io/library/jhlite:0.22.1-SNAPSHOT" not found
[INFO]     [creator]     ===> DETECTING
[INFO]     [creator]     6 of 14 buildpacks participating
[INFO]     [creator]     paketo-buildpacks/bellsoft-liberica 9.9.0-ea
[INFO]     [creator]     paketo-buildpacks/ca-certificates   3.4.0
[INFO]     [creator]     paketo-buildpacks/syft              1.22.1
[INFO]     [creator]     paketo-buildpacks/executable-jar    6.5.0
[INFO]     [creator]     paketo-buildpacks/spring-boot       5.20.0
[INFO]     [creator]     paketo-buildpacks/native-image      5.6.0
[INFO]     [creator]     ===> RESTORING
[INFO]     [creator]     ===> BUILDING
[INFO]     [creator]     
[INFO]     [creator]     Paketo Buildpack for BellSoft Liberica 9.9.0-ea
[INFO]     [creator]       https://github.com/paketo-buildpacks/bellsoft-liberica
[INFO]     [creator]       Build Configuration:
[INFO]     [creator]         $BP_JVM_JLINK_ARGS           --no-man-pages --no-header-files --strip-debug --compress=1  configure custom link arguments (--output must be omitted)
[INFO]     [creator]         $BP_JVM_JLINK_ENABLED        false                                                        enables running jlink tool to generate custom JRE
[INFO]     [creator]         $BP_JVM_TYPE                 JRE                                                          the JVM type - JDK or JRE
[INFO]     [creator]         $BP_JVM_VERSION              11                                                           the Java version
[INFO]     [creator]       Launch Configuration:
[INFO]     [creator]         $BPL_DEBUG_ENABLED           false                                                        enables Java remote debugging support
[INFO]     [creator]         $BPL_DEBUG_PORT              8000                                                         configure the remote debugging port
[INFO]     [creator]         $BPL_DEBUG_SUSPEND           false                                                        configure whether to suspend execution until a debugger has attached
[INFO]     [creator]         $BPL_HEAP_DUMP_PATH                                                                       write heap dumps on error to this path
[INFO]     [creator]         $BPL_JAVA_NMT_ENABLED        true                                                         enables Java Native Memory Tracking (NMT)
[INFO]     [creator]         $BPL_JAVA_NMT_LEVEL          summary                                                      configure level of NMT, summary or detail
[INFO]     [creator]         $BPL_JFR_ARGS                                                                             configure custom Java Flight Recording (JFR) arguments
[INFO]     [creator]         $BPL_JFR_ENABLED             false                                                        enables Java Flight Recording (JFR)
[INFO]     [creator]         $BPL_JMX_ENABLED             false                                                        enables Java Management Extensions (JMX)
[INFO]     [creator]         $BPL_JMX_PORT                5000                                                         configure the JMX port
[INFO]     [creator]         $BPL_JVM_HEAD_ROOM           0                                                            the headroom in memory calculation
[INFO]     [creator]         $BPL_JVM_LOADED_CLASS_COUNT  35% of classes                                               the number of loaded classes in memory calculation
[INFO]     [creator]         $BPL_JVM_THREAD_COUNT        250                                                          the number of threads in memory calculation
[INFO]     [creator]         $JAVA_TOOL_OPTIONS                                                                        the JVM launch flags
[INFO]     [creator]         Using Java version 17 extracted from MANIFEST.MF
[INFO]     [creator]       BellSoft Liberica NIK 17.0.5: Contributing to layer
[INFO]     [creator]         Downloading from https://download.bell-sw.com/vm/22.3.0/bellsoft-liberica-vm-core-openjdk17.0.5+8-22.3.0+1-ea-linux-amd64.tar.gz
[INFO]     [creator]         Verifying checksum
[INFO]     [creator]         Expanding to /layers/paketo-buildpacks_bellsoft-liberica/native-image-svm
[INFO]     [creator]         Adding 127 container CA certificates to JVM truststore
[INFO]     [creator]         Writing env.build/JAVA_HOME.override
[INFO]     [creator]         Writing env.build/JDK_HOME.override
[INFO]     [creator]     
[INFO]     [creator]     Paketo Buildpack for CA Certificates 3.4.0
[INFO]     [creator]       https://github.com/paketo-buildpacks/ca-certificates
[INFO]     [creator]       Launch Helper: Contributing to layer
[INFO]     [creator]         Creating /layers/paketo-buildpacks_ca-certificates/helper/exec.d/ca-certificates-helper
[INFO]     [creator]     
[INFO]     [creator]     Paketo Buildpack for Syft 1.22.1
[INFO]     [creator]       https://github.com/paketo-buildpacks/syft
[INFO]     [creator]         Downloading from https://github.com/anchore/syft/releases/download/v0.60.3/syft_0.60.3_linux_amd64.tar.gz
[INFO]     [creator]         Verifying checksum
[INFO]     [creator]         Writing env.build/SYFT_CHECK_FOR_APP_UPDATE.default
[INFO]     [creator]     
[INFO]     [creator]     Paketo Buildpack for Executable JAR 6.5.0
[INFO]     [creator]       https://github.com/paketo-buildpacks/executable-jar
[INFO]     [creator]       Class Path: Contributing to layer
[INFO]     [creator]         Writing env.build/CLASSPATH.delim
[INFO]     [creator]         Writing env.build/CLASSPATH.prepend
[INFO]     [creator]     
[INFO]     [creator]     Paketo Buildpack for Spring Boot 5.20.0
[INFO]     [creator]       https://github.com/paketo-buildpacks/spring-boot
[INFO]     [creator]       Build Configuration:
[INFO]     [creator]         $BP_SPRING_CLOUD_BINDINGS_DISABLED   false  whether to contribute Spring Boot cloud bindings support
[INFO]     [creator]       Launch Configuration:
[INFO]     [creator]         $BPL_SPRING_CLOUD_BINDINGS_DISABLED  false  whether to auto-configure Spring Boot environment properties from bindings
[INFO]     [creator]         $BPL_SPRING_CLOUD_BINDINGS_ENABLED   true   Deprecated - whether to auto-configure Spring Boot environment properties from bindings
[INFO]     [creator]       Class Path: Contributing to layer
[INFO]     [creator]         Writing env.build/CLASSPATH.append
[INFO]     [creator]         Writing env.build/CLASSPATH.delim
[INFO]     [creator]       Image labels:
[INFO]     [creator]         org.opencontainers.image.title
[INFO]     [creator]         org.opencontainers.image.version
[INFO]     [creator]         org.springframework.boot.version
[INFO]     [creator]     Warning: BOM table is deprecated in this buildpack api version, though it remains supported for backwards compatibility. Buildpack authors should write BOM information to <layer>.sbom.<ext>, launch.sbom.<ext>, or build.sbom.<ext>.
[INFO]     [creator]     
[INFO]     [creator]     Paketo Buildpack for Native Image 5.6.0
[INFO]     [creator]       https://github.com/paketo-buildpacks/native-image
[INFO]     [creator]       Build Configuration:
[INFO]     [creator]         $BP_BINARY_COMPRESSION_METHOD                Compression mechanism used to reduce binary size. Options: `none` (default), `upx` or `gzexe`
[INFO]     [creator]         $BP_NATIVE_IMAGE                       true  enable native image build
[INFO]     [creator]         $BP_NATIVE_IMAGE_BUILD_ARGUMENTS             arguments to pass to the native-image command
[INFO]     [creator]         $BP_NATIVE_IMAGE_BUILD_ARGUMENTS_FILE        a file with arguments to pass to the native-image command
[INFO]     [creator]         $BP_NATIVE_IMAGE_BUILT_ARTIFACT              the built application artifact explicitly, required if building from a JAR
[INFO]     [creator]       Native Image: Contributing to layer
[INFO]     [creator]         Executing native-image -H:+StaticExecutableWithDynamicLibC -H:Name=/layers/paketo-buildpacks_native-image/native-image/tech.jhipster.lite.JHLiteApp -cp /workspace:/workspace/BOOT-INF/classes:/workspace/BOOT-INF/lib/spring-boot-3.0.0-RC2.jar:/workspace/BOOT-INF/lib/spring-context-6.0.0-RC4.jar:/workspace/BOOT-INF/lib/spring-boot-autoconfigure-3.0.0-RC2.jar:/workspace/BOOT-INF/lib/logback-classic-1.4.4.jar:/workspace/BOOT-INF/lib/logback-core-1.4.4.jar:/workspace/BOOT-INF/lib/log4j-to-slf4j-2.19.0.jar:/workspace/BOOT-INF/lib/log4j-api-2.19.0.jar:/workspace/BOOT-INF/lib/jul-to-slf4j-2.0.3.jar:/workspace/BOOT-INF/lib/jakarta.annotation-api-2.1.1.jar:/workspace/BOOT-INF/lib/spring-core-6.0.0-RC4.jar:/workspace/BOOT-INF/lib/spring-jcl-6.0.0-RC4.jar:/workspace/BOOT-INF/lib/snakeyaml-1.33.jar:/workspace/BOOT-INF/lib/commons-lang3-3.12.0.jar:/workspace/BOOT-INF/lib/commons-io-2.11.0.jar:/workspace/BOOT-INF/lib/org.eclipse.jgit-6.3.0.202209071007-r.jar:/workspace/BOOT-INF/lib/JavaEWAH-1.1.13.jar:/workspace/BOOT-INF/lib/slf4j-api-2.0.3.jar:/workspace/BOOT-INF/lib/jackson-databind-2.14.0-rc3.jar:/workspace/BOOT-INF/lib/jackson-annotations-2.14.0-rc3.jar:/workspace/BOOT-INF/lib/jackson-core-2.14.0-rc3.jar:/workspace/BOOT-INF/lib/jackson-datatype-jdk8-2.14.0-rc3.jar:/workspace/BOOT-INF/lib/jackson-datatype-jsr310-2.14.0-rc3.jar:/workspace/BOOT-INF/lib/jackson-module-parameter-names-2.14.0-rc3.jar:/workspace/BOOT-INF/lib/tomcat-embed-core-10.1.1.jar:/workspace/BOOT-INF/lib/tomcat-embed-websocket-10.1.1.jar:/workspace/BOOT-INF/lib/spring-web-6.0.0-RC4.jar:/workspace/BOOT-INF/lib/spring-beans-6.0.0-RC4.jar:/workspace/BOOT-INF/lib/spring-webmvc-6.0.0-RC4.jar:/workspace/BOOT-INF/lib/spring-aop-6.0.0-RC4.jar:/workspace/BOOT-INF/lib/spring-expression-6.0.0-RC4.jar:/workspace/BOOT-INF/lib/springdoc-openapi-starter-webmvc-ui-2.0.0-RC1.jar:/workspace/BOOT-INF/lib/swagger-ui-4.15.0.jar:/workspace/BOOT-INF/lib/webjars-locator-core-0.52.jar:/workspace/BOOT-INF/lib/classgraph-4.8.149.jar:/workspace/BOOT-INF/lib/springdoc-openapi-starter-webmvc-api-2.0.0-RC1.jar:/workspace/BOOT-INF/lib/springdoc-openapi-starter-common-2.0.0-RC1.jar:/workspace/BOOT-INF/lib/swagger-core-jakarta-2.2.4.jar:/workspace/BOOT-INF/lib/swagger-annotations-jakarta-2.2.4.jar:/workspace/BOOT-INF/lib/swagger-models-jakarta-2.2.4.jar:/workspace/BOOT-INF/lib/jackson-dataformat-yaml-2.14.0-rc3.jar:/workspace/BOOT-INF/lib/tomcat-embed-el-10.1.1.jar:/workspace/BOOT-INF/lib/hibernate-validator-8.0.0.Final.jar:/workspace/BOOT-INF/lib/jakarta.validation-api-3.0.2.jar:/workspace/BOOT-INF/lib/jboss-logging-3.5.0.Final.jar:/workspace/BOOT-INF/lib/classmate-1.5.1.jar:/workspace/BOOT-INF/lib/spring-boot-actuator-autoconfigure-3.0.0-RC2.jar:/workspace/BOOT-INF/lib/spring-boot-actuator-3.0.0-RC2.jar:/workspace/BOOT-INF/lib/micrometer-observation-1.10.0.jar:/workspace/BOOT-INF/lib/micrometer-commons-1.10.0.jar:/workspace/BOOT-INF/lib/micrometer-core-1.10.0.jar:/workspace/BOOT-INF/lib/HdrHistogram-2.1.12.jar:/workspace/BOOT-INF/lib/LatencyUtils-2.0.3.jar:/workspace/BOOT-INF/lib/maven-model-3.8.6.jar:/workspace/BOOT-INF/lib/plexus-utils-3.3.1.jar:/workspace/BOOT-INF/lib/joox-1.6.2.jar:/workspace/BOOT-INF/lib/jaxb-api-2.3.0.jar:/workspace/BOOT-INF/lib/dom4j-2.1.3.jar:/workspace/BOOT-INF/lib/mongodb-driver-sync-4.8.0-beta0.jar:/workspace/BOOT-INF/lib/bson-4.8.0-beta0.jar:/workspace/BOOT-INF/lib/mongodb-driver-core-4.8.0-beta0.jar:/workspace/BOOT-INF/lib/bson-record-codec-4.8.0-beta0.jar:/workspace/BOOT-INF/lib/spring-data-mongodb-4.0.0-RC2.jar:/workspace/BOOT-INF/lib/spring-tx-6.0.0-RC4.jar:/workspace/BOOT-INF/lib/spring-data-commons-3.0.0-RC2.jar:/workspace/BOOT-INF/lib/mongock-springboot-v3-5.2.0.jar:/workspace/BOOT-INF/lib/mongock-runner-core-5.1.6.jar:/workspace/BOOT-INF/lib/mongock-driver-api-5.1.6.jar:/workspace/BOOT-INF/lib/javassist-3.28.0-GA.jar:/workspace/BOOT-INF/lib/guava-30.1.1-jre.jar:/workspace/BOOT-INF/lib/failureaccess-1.0.1.jar:/workspace/BOOT-INF/lib/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:/workspace/BOOT-INF/lib/checker-qual-3.8.0.jar:/workspace/BOOT-INF/lib/error_prone_annotations-2.5.1.jar:/workspace/BOOT-INF/lib/j2objc-annotations-1.3.jar:/workspace/BOOT-INF/lib/javax.inject-1.jar:/workspace/BOOT-INF/lib/reflections-0.10.1.jar:/workspace/BOOT-INF/lib/maven-artifact-3.6.1.jar:/workspace/BOOT-INF/lib/gson-2.9.1.jar:/workspace/BOOT-INF/lib/mongock-api-5.1.6.jar:/workspace/BOOT-INF/lib/mongock-utils-5.1.6.jar:/workspace/BOOT-INF/lib/mongodb-springdata-v4-driver-5.2.0.jar:/workspace/BOOT-INF/lib/mongock-driver-core-5.1.6.jar:/workspace/BOOT-INF/lib/mongodb-sync-v4-driver-5.1.6.jar:/workspace/BOOT-INF/lib/svm-19.2.1.jar:/workspace/BOOT-INF/lib/graal-sdk-19.2.1.jar:/workspace/BOOT-INF/lib/objectfile-19.2.1.jar:/workspace/BOOT-INF/lib/pointsto-19.2.1.jar:/workspace/BOOT-INF/lib/truffle-nfi-19.2.1.jar:/workspace/BOOT-INF/lib/truffle-api-19.2.1.jar:/workspace/BOOT-INF/lib/compiler-19.2.1.jar:/workspace/BOOT-INF/lib/jakarta.xml.bind-api-4.0.0.jar:/workspace/BOOT-INF/lib/jakarta.activation-api-2.1.0.jar:/workspace/BOOT-INF/lib/objenesis-3.2.jar:/workspace/BOOT-INF/lib/jsr305-3.0.2.jar:/workspace/BOOT-INF/lib/spring-boot-jarmode-layertools-3.0.0-RC2.jar tech.jhipster.lite.JHLiteApp
[INFO]     [creator]     Fatal error: java.lang.IllegalAccessError: failed to access class com.oracle.svm.core.genscavenge.HeapChunk$Options from class com.oracle.svm.core.genscavenge.HeapChunk_OptionDescriptors (com.oracle.svm.core.genscavenge.HeapChunk$Options is in unnamed module of loader 'app'; com.oracle.svm.core.genscavenge.HeapChunk_OptionDescriptors is in unnamed module of loader com.oracle.svm.hosted.NativeImageClassLoaderSupport$ClassPathClassLoader @224edc67)
[INFO]     [creator]     	at com.oracle.svm.core.genscavenge.HeapChunk_OptionDescriptors.get(HeapChunk_OptionDescriptors.java:17)
[INFO]     [creator]     	at com.oracle.svm.core.genscavenge.HeapChunk_OptionDescriptors$1.next(HeapChunk_OptionDescriptors.java:42)
[INFO]     [creator]     	at com.oracle.svm.core.genscavenge.HeapChunk_OptionDescriptors$1.next(HeapChunk_OptionDescriptors.java:33)
[INFO]     [creator]     	at com.oracle.svm.common.option.CommonOptionParser.collectOptions(CommonOptionParser.java:169)
[INFO]     [creator]     	at com.oracle.svm.core.option.SubstrateOptionsParser.collectOptions(SubstrateOptionsParser.java:116)
[INFO]     [creator]     	at com.oracle.svm.hosted.option.HostedOptionParser.collectOptions(HostedOptionParser.java:63)
[INFO]     [creator]     	at com.oracle.svm.hosted.option.HostedOptionParser.<init>(HostedOptionParser.java:58)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageClassLoaderSupport.setupHostedOptionParser(NativeImageClassLoaderSupport.java:186)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageGeneratorRunner.installNativeImageClassLoader(NativeImageGeneratorRunner.java:181)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:123)
[INFO]     [creator]     	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:610)
[INFO]     [creator]     Error: Image build request failed with exit status 1
[INFO]     [creator]     unable to invoke layer creator
[INFO]     [creator]     unable to contribute native-image layer
[INFO]     [creator]     error running build
[INFO]     [creator]     exit status 1
[INFO]     [creator]     ERROR: failed to build: exit status 1
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:42 min
[INFO] Finished at: 2022-11-11T16:18:49+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.springframework.boot:spring-boot-maven-plugin:3.0.0-RC2:build-image (default-cli) on project jhlite: Execution default-cli of goal org.springframework.boot:spring-boot-maven-plugin:3.0.0-RC2:build-image failed: Builder lifecycle 'creator' failed with status code 51 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

**Comment by atomfrede** [comment]
Let me check. I always tested the native executable created with `./mvnw native:compile -Pnative -DskipTests`. But I am pretty sure I also build the image at least once.

**Comment by atomfrede** [comment]
Same for me. I checked the minimal app from spring initializer and it worked. Will do some investigations.

**Comment by atomfrede** [comment]
I think I have found the issue. Need to do some more testing and building the image later.

**Comment by atomfrede** [comment]
@pascalgrimaud Can you try again? It took some time as the build took too long on the macs and on my linux workstation the frontend build didn't work. I tried both the native image and the native executable, both work fine.

**Comment by pascalgrimaud** [comment]
Thanks @atomfrede : I'll try it tomorrow and if it's ok, we're ready for this awesome release

File: README.md [hunk 7]
```diff
@@ -78,6 +78,53 @@ To launch tests and integration tests:
...
+To create the image, run the following goal:
+
+```
+./mvnw spring-boot:build-image -Pnative
+```
+
+Then, you can run the app like any other container:
+
+```
+docker run --rm docker.io/library/jhlite:<VERSION>
```
Comment (pascalgrimaud) [inline README.md:98]:
```suggestion
docker run -p 7471:7471 --rm docker.io/library/jhlite:<VERSION>
```

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Review by pascalgrimaud — APPROVED** [review]

**Comment by atomfrede** [comment]
Bounty claimed https://opencollective.com/generator-jhipster/expenses/107436

**Comment by pascalgrimaud** [comment]
@atomfrede : approved

**Pull request merged**